services/graph/pkg/identity/ldap_education_class.go

```diff
@@ -139,6 +140,106 @@ func (i *LDAP) DeleteEducationClass(ctx context.Context, id string) error {
 	return nil
 }
 
+// UpdateEducationClass implements the EducationBackend interface for the LDAP backend.
+// Only the displayName and externalID are supported to change at this point.
+func (i *LDAP) UpdateEducationClass(ctx context.Context, id string, class libregraph.EducationClass) (*libregraph.EducationClass, error) {
+	logger := i.logger.SubloggerWithRequestID(ctx)
+	logger.Debug().Str("backend", "ldap").Msg("UpdateEducationClass")
+	if !i.writeEnabled {
+		return nil, ErrReadOnly
+	}
+
+	g, err := i.getLDAPGroupByID(id, false)
+	if err != nil {
+		return nil, err
+	}
+
+	var updateNeeded bool
+
+	if class.GetId() != "" {
+		if g.GetEqualFoldAttributeValue(i.groupAttributeMap.id) != class.GetId() {
+			return nil, errorcode.New(errorcode.NotAllowed, "changing the GroupID is not allowed")
+		}
+	}
+
+	if class.GetOnPremisesSamAccountName() != "" {
+		return nil, errorcode.New(errorcode.NotSupported, "changing the SAM account name is currently not supported")
+	}
+
+	if class.GetOnPremisesDomainName() != "" {
+		return nil, errorcode.New(errorcode.NotSupported, "changing the SAM account name is currently not supported")
+	}
```

Comment:
Ugh. Seems we need to bump the libregraph-api module.`onPremisesSamAccountName` and `onPremisesDomainName` were removed from `educationClass`  recently. Could be please remove the code touching these two attributes? Sorry for that.

Reply:
These have been removed.